src/ashley/machina_extensions/forum_permission/handler.py

```diff
@@ -41,5 +45,7 @@ def forum_list_filter(self, qs, user):
         """
         forums_to_show = super().forum_list_filter(qs, user)
```

Comment:
Extra feedback, It makes me wonder if there is no high-level method that would need to be overridden in Django's Machina to be sure deleted forums don't appear. I'm thinking for instance of _get_all_forums from Django's Machina. Maybe it would be a good idea to add a filter directly on a high-level method to prevent errors.
see machina/apps/forum_permission/handler.py

Reply:
Thanks for your feedback!

I don't think we should set this behavior in `_get_all_forums()` because the purpose of this function is to return all the forums, without worrying about permissions, with a caching mechanism. What if we want administrators to be able to see archived forums, for example? It would not be possible if we introduce this behavior in this method.

In addition to that, this function is not used everywhere, there is no high-level method used everywhere.
Multiple django-machina components are directly using `Forum.objects.all()` and apply the `forum_list_filter` function on the result. That's why I chose to override this function.

I agree with you: this could be better. But it implies overriding a lot of django machina components to introduce a high-level method to handle this :/